server/test/controllers/admin/AdminProgramBlocksControllerTest.java

```diff
@@ -99,6 +110,74 @@ public void create_withProgram_addsRepeatedBlock() {
     assertThat(program.getProgramDefinition().blockDefinitions()).hasSize(3);
   }
 
+  @Test
+  public void view_throwsError_ifCalledOnNoneActiveProgram() throws Exception {
+    Request request = addCSRFToken(Helpers.fakeRequest()).build();
+    Program program = ProgramBuilder.newDraftProgram("test program").build();
+
+    assertThatThrownBy(() -> controller.view(request, program.id, 1L))
+        .isInstanceOf(NotViewableException.class);
+  }
+
+  @Test
+  public void view_withInvalidProgram_notFound() {
+    Request request = fakeRequest().build();
+    assertThatThrownBy(() -> controller.view(request, 1L, 1L))
+        .isInstanceOf(NotViewableException.class);
+  }
+
+  @Test
+  public void view_withInvalidBlock_notFound() {
+    Program program = ProgramBuilder.newActiveProgram().build();
+    Request request = fakeRequest().build();
+    Result result = controller.view(request, program.id, 2L);
+
+    assertThat(result.status()).isEqualTo(NOT_FOUND);
+  }
+
+  @Test
+  public void view_withProgram_OK()
+      throws UnsupportedQuestionTypeException, InvalidUpdateException {
+    Program program =
+        ProgramBuilder.newActiveProgram("Public name", "Public description")
+            // Override only admin name and description to distinguish from applicant-visible
+            // name/description.
+            .withName("Admin name")
+            .withDescription("Admin description")
+            .build();
+    Question appName = testQuestionBank.applicantName();
+    appName.save();
+    Request request = addCSRFToken(fakeRequest()).build();
+    Result result = controller.view(request, program.id, 1L);
+
+    assertThat(result.status()).isEqualTo(OK);
+    String html = Helpers.contentAsString(result);
+    assertThat(html).contains(appName.getQuestionDefinition().getQuestionText().getDefault());
+    assertThat(html).contains(appName.getQuestionDefinition().getQuestionHelpText().getDefault());
+    assertThat(html).contains("Admin ID: " + appName.getQuestionDefinition().getName());
+    assertThat(html)
+        .contains("Public name")
+        .contains("Public description")
+        .contains("Admin description")
+        // Similar to program index page we don't show admin name.
+        .doesNotContain("Admin name");
+
+    QuestionDefinition questionDefinition =
```

Comment:
**suggestion (non-blocking):** This is a bit vague as it's not clear the utility of this question.  Suggest naming something like otherQuestionDef, differentQuestionDef